### ISSUE TITLE
Google Sheets Destinations Scoping

### ISSUE THREAD

**Issue by misteryeo** [airbytehq/airbyte#11264]
**Requirements**
We'd like to implement a Google Sheets destination as there are multiple requests from our community (eg. https://github.com/airbytehq/airbyte/issues/4650) and we believe this will help activation on our Cloud offering.

We'd like this destination to be written in Python (see [destination-kvdb](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/destination-kvdb) for an example). The destination should support the overwrite and append sync modes. append_dedupe is not in scope for this implementation. 

Please scope/estimate the technical work required for this destination, including any OAuth requirements, and the work required to deploy this destination to cloud.

**Expected output**
Please share a Google Doc and tag @sherifnada @misteryeo once scoping is complete.

**Comment by bazarnov** [comment]
@misteryeo I'm aware of these limitations, so:
> Google Sheets [has limits](https://gsuitetips.com/tips/sheets/google-spreadsheet-limitations/) so we might need to consider some sort of cap of replication loads.

This could be checked and properly configured based on incoming data dimensions, before making any uploads to the destination, some sort of intermediate staging could also be used. Will see once the implementation takes place.

> Another consideration is around syncing from a single table that creates one sheet vs multiple tables and multiple sheets.

Yes, this could be potential long-run back-and-forth implementing. This is discussible anyway, i think this is the biggest question for now.

**Comment by bazarnov** [comment]
@misteryeo @sherifnada 
Here is the DRAFT of the scoping, please note that we don't have any example of any destination scoping either Java or Python so far, so i did it how i see it. Feel free to correct anything up there and comment out some parts once needed. Thanks. https://docs.google.com/document/d/1vNjcgwRrqSu4OHoGYMcXb6DjaYG8Hv0isCQMUAHyShs/edit

**Comment by bazarnov** [comment]
https://github.com/airbytehq/airbyte/issues/11918
